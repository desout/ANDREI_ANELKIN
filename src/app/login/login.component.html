<div class="container">
  <ng-container *ngIf="!(userService.currentUser$ | async)?.name;then isNotLoggedTemplate; else isLoggedTemplate"></ng-container>
<ng-template #isNotLoggedTemplate>
  <h1>{{ 'LOGIN.TITLE' | translate }}</h1>
  <form (ngSubmit)="onSubmit()" [formGroup]="loginForm" #formDir="ngForm">
    <div class="form-group">
      <label>{{'INFO.USERNAME' | translate}}
        <input class="form-control" formControlName="userName">
      </label>
      <div *ngIf="userName.pending">{{'ERROR.VALIDATING' | translate}}</div>
      <div *ngIf="userName.invalid && (userName.dirty || userName.touched)"
           class="alert alert-danger">
        <div *ngIf="userName.errors['required']">
          {{'INFO.USERNAME' | translate}} {{'ERROR.REQUIRED' | translate}}.
        </div>
        <div *ngIf="userName.errors['isBadName']">
          {{'ERROR.USERNOTFOUND' | translate}}
        </div>
      </div>
    </div>

    <div class="form-group">
      <label>{{'INFO.PASSWORD' | translate}}
        <input type="password" class="form-control" formControlName="password">
      </label>
      <div *ngIf="password.invalid && (password.dirty || password.touched)"
           class="alert alert-danger">
        <div *ngIf="password.errors['required']">
          {{'INFO.PASSWORD' | translate}} {{'ERROR.REQUIRED' | translate}}.
        </div>
        <div *ngIf="password.errors['minlength']">
          {{'ERROR.MINLENGTH' | translate}}} 4
        </div>
        <div *ngIf="password.errors['badPassword']">
          {{'ERROR.BADPASSWORDLOGIN' | translate}}
        </div>
      </div>
    </div>
    <button type="submit" class="btn btn-default" [disabled]="loginForm.invalid">{{'INFO.SUBMITBTN' | translate}}</button>
    <a routerLink="/account/forgotPassword" class="btn btn-info">{{'INFO.FORGOTPASSWBTN' | translate}}</a>
  </form>
</ng-template>
<ng-template #isLoggedTemplate>
  <h1>{{ 'LOGIN.LOGGED' | translate }}</h1>
  <button (click)="onClickLogout()" class="btn btn-danger align-self-lg-center"> {{'INFO.LOGOUT' | translate}}</button>
</ng-template>
</div>
