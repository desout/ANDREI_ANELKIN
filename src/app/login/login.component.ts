import {Component, OnInit} from '@angular/core';
import {FormControl, FormGroup, Validators} from '@angular/forms';
import {UserService} from '../user.service';
import {AuthService, ResponseType, UserLogin} from '../auth.service';
import {map} from 'rxjs/operators';
import {Router} from '@angular/router';
import {NgbModal} from '@ng-bootstrap/ng-bootstrap';
import {PopUpModelComponent} from '../pop-up-model/pop-up-model.component';

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss']
})
export class LoginComponent implements OnInit {
  user: UserLogin = { name: '', password: ''};
  loginForm: FormGroup;

  constructor(private modalService: NgbModal,
              private router: Router,
              protected userService: UserService,
              private authService: AuthService) {
  }

  ngOnInit(): void {
    this.loginForm = new FormGroup({
      userName: new FormControl(this.user.name, {
        validators: [Validators.required]
      }),
      password: new FormControl(this.user.password, {validators: [Validators.required, Validators.minLength(4)]})
    });
  }

  onSubmit() {
    this.authService.login({name: this.userName.value, password: this.password.value} as UserLogin).pipe(map(response => {
      if (!(response as ResponseType).success) {
        this.loginForm.get('password').setErrors({
          'badPassword': true
        });
      } else {
        this.openPopUp();
        this.router.navigate(['./infoTab']);
      }
    })).subscribe();
  }
  onClickLogout() {
    this.authService.logout();
  }
  openPopUp() {
    const modalRef = this.modalService.open(PopUpModelComponent);
    modalRef.componentInstance.name = this.userName.value;
  }
  get userName() {
    return this.loginForm.get('userName');
  }

  get password() {
    return this.loginForm.get('password');
  }
}
