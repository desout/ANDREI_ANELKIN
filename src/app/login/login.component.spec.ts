import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginComponent } from './login.component';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TestModule} from '../Test.module';

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [TestModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
