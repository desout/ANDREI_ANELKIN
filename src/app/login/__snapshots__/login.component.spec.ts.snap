// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should create 1`] = `
<div
  id="root0"
  ng-version="7.0.3"
>
  <div
    class="container"
  >
    
    <h1>
      LOGIN.TITLE
    </h1>
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <div
        class="form-group"
      >
        <label>
          INFO.USERNAME 
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="userName"
            ng-reflect-name="userName"
          />
        </label>
        
        
      </div>
      <div
        class="form-group"
      >
        <label>
          INFO.PASSWORD 
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="password"
            ng-reflect-name="password"
            type="password"
          />
        </label>
        
      </div>
      <button
        class="btn btn-default"
        disabled=""
        type="submit"
      >
        INFO.SUBMITBTN
      </button>
      <a
        class="btn btn-info"
        href="/account/forgotPassword"
        ng-reflect-router-link="/account/forgotPassword"
        routerlink="/account/forgotPassword"
      >
        INFO.FORGOTPASSWBTN
      </a>
    </form>
    
    
  </div>
</div>
`;
